.github/workflows/rpm-build-and-test.yml

```diff
@@ -0,0 +1,688 @@
+      - name: Test repo
+        id: test-repo
+        if: steps.create-repo.outcome == 'success'
+        continue-on-error: true
+        run: . ci/gha_functions.sh;
+             dnf --disablerepo=\* --repofrompath
+                 testrepo,file://${REPO_PATH}${{ github.run_number }}/artifact/artifacts/$TARGET
+                 repoquery -a
+      - name: Remove lastSuccessfulBuild link and exit failure
+        if: steps.test-repo.outcome != 'success'
+        run: rm -f ${REPO_PATH}lastSuccessfulBuild;
+             exit 1
+      - name: Publish RPMs
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ env.DISTRO_NAME }} ${{ env.DISTRO_VERSION }} RPM repository
+          path: ${{ env.REPO_PATH}}${{ github.run_number }}/artifact/artifacts/${{ env.TARGET }}
+
+  Calc-functional-matrix:
+    name: Calculate Functional Testing Matrix
+    runs-on: [self-hosted, wolf]
+    needs: [Import-commit-pragmas]
+    if: needs.Import-commit-pragmas.outputs.run-gha == 'true' &&
+        ((!cancelled()) || success() || failure())
+    outputs:
+      matrix: ${{ steps.matrix.outputs.text }}
+    steps:
+      - name: Import commit pragmas
+        uses: ./.github/actions/import-commit-pragmas
+      - name: Calculate Functional Testing Matrix
+        id: matrix
+        run: |  # do not use the non-| format for this script
+             l=()
+             trap 'echo "text=[$(IFS=","; echo "${l[*]}")]" >> $GITHUB_OUTPUT' EXIT
+             if ${CP_SKIP_FUNC_TEST:-false}; then
+                 exit 0
+             fi
+             # it might seem tempting to factor in the result of the build for this
+             # distro here and not include a failed build in the test matrix
+             # the problem with that is that if/when the user asks GHA to rebuild
+             # all faiiled jobs and a previously failed RPM job is successful, the
+             # test matrix won't include testing it since it was calculated and was
+             # successful on the previous run without the failed build stage in it
+             if ! ${CP_SKIP_BUILD_EL8_RPM:-false} &&
+                ! ${CP_SKIP_FUNC_TEST_EL8:-false}; then
+                 # it would definitely be nicer to get these into the environment
+                 # as unquoted strings so that we didn't have to double quote here
+                 l+=('"el8"')
+             fi
+             if ! ${CP_SKIP_BUILD_EL9_RPM:-false} &&
+                ! ${CP_SKIP_FUNC_TEST_EL9:-false}; then
+                 l+=('"el9"')
+             fi
+             if ${{ github.event_name == 'push' }} ||
+                (${{ github.event_name == 'pull_request' }} &&
+                ! ${CP_SKIP_BUILD_LEAP15_RPM:-false} &&
+                ! ${CP_SKIP_FUNC_TEST_LEAP15:-true}); then
+                 l+=('"leap15"')
+             fi
+
+  Functional:
+    name: Functional Testing
+    runs-on: [self-hosted, wolf]
+    timeout-minutes: 7200
```

Comment:
Shouldn't this value be a sum of all the step timeouts?  The `Request and Provision a Cluster` and `Run Test` steps each define a 5 day timeout.

Reply:
Not necessarily.  They are all just supposed to be "big enough" values that our jobs, running normally should not timeout.

Given that we'd not expect any of the steps to take anywhere near this value, we'd not really want to let the whole job be a sum of the step.

Perhaps the steps need a bit of adjustment/tuning at some point.